anchor/common/qbft/src/lib.rs

```diff
@@ -696,11 +696,18 @@ where
             return;
         }
 
-        // Make sure that we have accepted a proposal for this round
-        if !self.proposal_accepted_for_current_round {
-            warn!(from=?operator_id, ?self.state, "Have not accepted Proposal for current round yet");
+        // If we have accepted a proposal, check that the commit matches it
+        // But allow commits without proposal (for catch-up scenarios)
+        if self.proposal_accepted_for_current_round {
```

Comment:
Do we need `proposal_accepted_for_current_round` at all, or could we use `self.proposal_root`?

Reply:
Why has this been resolved?

Reply:
We can leave it like this for now, but consider if we can use only `proposal_root`. Wdyt @dknopik ?